subiquity/ui/views/ubuntu_pro.py

```diff
@@ -125,76 +197,259 @@ def cancel(self, sender) -> None:
-    title = _("Enable Ubuntu Pro")
-    excerpt = _("Enter your Ubuntu Pro token if you want to enroll "
-                "this system.")
+    title = _("Upgrade to Ubuntu Pro")
 
     def __init__(self, controller, token: str):
         """ Initialize the view with the default value for the token. """
         self.controller = controller
 
-        self.form = UbuntuProForm(initial={"token": token})
-
-        def on_cancel(_: UbuntuProForm):
+        self.upgrade_yes_no_form = UpgradeYesNoForm(initial={
+            "skip": not token,
+            "upgrade": bool(token),
+            })
+        self.upgrade_mode_form = UpgradeModeForm(initial={
+            "with_contract_token_subform": {"token": token},
+            })
+
+        def on_upgrade_yes_no_cancel(unused: UpgradeYesNoForm):
+            """ Function to call when hitting Done from the upgrade/skip
+            screen. """
             self.cancel()
 
-        connect_signal(self.form, 'submit', self.done)
-        connect_signal(self.form, 'cancel', on_cancel)
+        def on_upgrade_mode_cancel(unused: UpgradeModeForm):
+            """ Function to call when hitting Back from the contract token
+            form. """
+            self._w = self.upgrade_yes_no_screen()
+
+        connect_signal(self.upgrade_yes_no_form,
+                       'submit', self.upgrade_yes_no_done)
+        connect_signal(self.upgrade_yes_no_form,
+                       'cancel', on_upgrade_yes_no_cancel)
+        connect_signal(self.upgrade_mode_form,
+                       'submit', self.upgrade_mode_done)
+        connect_signal(self.upgrade_mode_form,
+                       'cancel', on_upgrade_mode_cancel)
+
+        super().__init__(self.upgrade_yes_no_screen())
+
+    def upgrade_mode_screen(self) -> Widget:
+        """ Return a screen that asks the user for his information (e.g.,
+        contract token).
+        +---------------------------------------------------------+
+        | To upgrade to Ubuntu Pro, you can enter your token      |
+        | manually.                                               |
+        |                                                         |
+        | [ How to Register -> ]                                  |
+        |                                                         |
+        | (X)  Add token manually                                 |
+        |      Token: C123456789ABCDEF                            |
+        |             This is your Ubuntu Pro token               |
+        |                                                         |
+        |                        [ Continue ]                     |
+        |                        [ Back     ]                     |
+        +---------------------------------------------------------+
+        """
+
+        excerpt = _("To upgrade to Ubuntu Pro, you can enter your token"
+                    " manually.")
+
+        how_to_register_btn = menu_btn(
+                _("How to Register"),
```

Comment:
Lowercase 'r' :)

Reply:
Ah, good catch! :eye: 

According to the mockups, this one should be uppercase ; but the one in the "How to register" screen title should be lowercase.

I will fix it now and ask Elio to confirm is this is what we want.

Reply:
Screenshot updated

Reply:
Ah yes, I was looking at the dialog title.
Good idea to ask about the ambiguity.